src/main/java/org/mbari/m3/dive/visualizer/AnnotationServiceHelper.java

```diff
@@ -0,0 +1,256 @@
+            }
+        }
+        return allDiveVideos;
+    }
+
+    // i am looking for the video_uuid that has the .mov video file, not .mp4
+    public String getVideoReferenceUUID(String video_uuid){
+        if(this.allAnnotationData.isJsonNull()) {
+            System.out.println("EMPTY ANNOTATION TREE - AnnotationServieHelper.getVideoReferenceUUID()");
+            return null;
+        }
+
+        // loop through the media
+        JsonArray allMedia = getMedia();
+        String video_reference_uuid = "";
+
+        for(int i = 0; i < allMedia.size(); i++) {   
+            // i have link, must get the uuid, now get the video_reference_uuid. 
+            String curr_uuid = allMedia.get(i).getAsJsonObject().get("video_uuid").toString();
+            if(video_uuid.equals(curr_uuid)){
+                //System.out.println(video_uuid + " matched " + curr_uuid);
+                // need to check the uri file name
+                String curr_uri = allMedia.get(i).getAsJsonObject().get("uri").toString();
+                if(curr_uri.charAt(curr_uri.length()-2)=='v') {
+                    //System.out.println("current uri.mov " + curr_uri);
+                    //System.out.println(".mov video_reference_uuid: " + allMedia.get(i).getAsJsonObject().get("video_reference_uuid").toString());
+                    video_reference_uuid = allMedia.get(i).getAsJsonObject().get("video_reference_uuid").toString();
+                    break;
+                }
+            }
+        }
+        return video_reference_uuid;
+    }
+
+    // This function takes the video_reference_uuid. It will return a list (Gson list) of annotations
+    public JsonArray getAnnotationsByVideoReferenceUUID(String video_reference_uuid){
+        if(this.allAnnotationData.isJsonNull()) {
+            System.out.println("EMPTY ANNOTATION TREE - AnnotationServieHelper.getAnnotationsByVideoReferenceUUID()");
+            return null;
+        }
+
+        JsonArray allAnnotations = getAnnotations();
+        JsonArray videoAnnotations = new JsonArray();
+        for(int i = 0; i < allAnnotations.size(); i++) {
+            if(video_reference_uuid.equals(allAnnotations.get(i).getAsJsonObject().get("video_reference_uuid").toString())){
+                videoAnnotations.add(allAnnotations.get(i).getAsJsonObject());
+            }
+        }
+        return videoAnnotations;
+    }
+
+    public JsonObject getVideoLinksAndAnnotations(){
+        if(this.allAnnotationData.isJsonNull()) {
+            System.out.println("EMPTY ANNOTATION TREE - AnnotationServieHelper.linksAndAnnotations()");
+            return null;
+        }
+        JsonArray allAnnotations = getAnnotations();
+        JsonArray allMedia = getMedia();
+        JsonArray videoLinks = getVideoLinks();
+        JsonObject linksAndUUID = new JsonObject();
+
+        // ================================================================================================ //
```

Comment:
In general for comments we should try to make them professional. This includes things like grammar, capitalization. As far as commenting the actual code, too much comments can make the code messy to read. Comments should be added to explain something which may not seem clear from reading the code. Your code is overall easy to follow, explaining each step in the code is not always necessary.